Classes/Utility/Theme.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  Theme.swift
+//  Freetime
+//
+//  Created by Nathan Tannar on 2018-12-15.
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+
+private let kPreferredThemeRawValueKey = "kPreferredThemeRawValueKey"
+
+enum Theme: Int {
+    case light = 0
+    case dark = 1
+
+    static var `default`: Theme {
+        let rawValue = UserDefaults.standard.integer(forKey: kPreferredThemeRawValueKey)
+        return Theme(rawValue: rawValue) ?? .light
+    }
+}
+
+final class Appearance {
+    private(set) static var currentTheme: Theme = .default
+
+    static func setCurrentTheme(_ theme: Theme, animated: Bool) {
+        UIView.animate(
+            withDuration: animated ? 0.5 : 0.0,
+            delay: 0,
+            usingSpringWithDamping: 1,
+            initialSpringVelocity: 0,
+            options: [.transitionCrossDissolve],
+            animations: {
+                Appearance.currentTheme = theme
+                UserDefaults.standard.set(theme.rawValue, forKey: kPreferredThemeRawValueKey)
+                NotificationCenter.default.post(name: kThemeDidChange, object: theme)
+        })
+    }
+}
+
+private let kThemeDidChange = NSNotification.Name("kThemeDidChange")
+
+class ThemeChangeNotifier: NSObject {
+    private let block: (_ notification: Notification?) -> Void
+
+    init(_ block: @escaping (_ notification: Notification?) -> Void) {
+        self.block = block
+        super.init()
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(notificationReceived(_:)),
+            name: kThemeDidChange,
+            object: nil
+        )
+    }
+
+    @objc private func notificationReceived(_ notification: Notification?) {
+        block(notification)
+    }
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+}
+
+protocol ThemeChangeListener: AnyObject {
```

Comment:
Why not putting `ThemeChangeListener` to a separate file?

Reply:
That was the plan, and for all your suggestions about this file. II didn't want to invest too heavy in formatting these if architecturally it was decided to use another method of theming